src/routers/upload.py

```diff
@@ -41,19 +45,142 @@ def format_size(size: int) -> str:
+        file_size=target_path.stat().st_size,
+        sha256=sha256,
+        bbox=bounds,
+        status=StatusEnum.pending,
+        user_id=user_id,
+        copy_time=copy_time,
+    )
+    dataset = Dataset(**data)
+
+    with use_client() as client:
+        try:
+            send_data = {
+                k: v
+                for k, v in dataset.model_dump().items()
+                if k != "id" and v is not None
+            }
+            response = client.table(settings.datasets_table).insert(send_data).execute()
+        except Exception as e:
+            logger.exception(f"Error uploading dataset: {str(e)}")
+            raise HTTPException(
+                status_code=400, detail=f"Error uploading dataset: {str(e)}"
+            )
+
+    return Dataset(**response.data[0])
+
+
+@router.post("/datasets/chunk")
+async def upload_geotiff_chunk(
+    file: UploadFile,
+    chunk: Annotated[int, Form()],
+    chunks: Annotated[int, Form()],
+    filename: Annotated[str, Form()],
+    copy_time: Annotated[int, Form()],
+    upload_id: Annotated[str, Form()],
+    token: Annotated[str, Depends(oauth2_scheme)],
+):
+    """
+    Handle chunked upload of a GeoTIFF file.
+
+    This endpoint receives chunks of a file, saves them temporarily,
+    and combines them when all chunks are received.
+    """
+    # Verify the token
+    user = verify_token(token)
+    if not user:
+        raise HTTPException(status_code=401, detail="Invalid token")
+
+    # Define the path for temporary chunk storage
+    temp_dir = Path(settings.tmp_upload_path) / upload_id
+    temp_dir.mkdir(parents=True, exist_ok=True)
+
+    # Save the chunk
+    chunk_file = temp_dir / f"chunk_{chunk}"
+    with chunk_file.open("wb") as buffer:
+        content = await file.read()
+        buffer.write(content)
+
+    # If this is the last chunk, combine all chunks
+    if int(chunk) == int(chunks) - 1:
+        combined_file = await combine_chunks(temp_dir, chunks, filename)
+
+        # Process the combined file (similar to the original upload endpoint)
+        uid = str(uuid.uuid4())
+        file_name = f"{uid}_{Path(filename).stem}.tif"
+        target_path = settings.archive_path / file_name
+
+        # Move the combined file to the target path
+        os.rename(combined_file, target_path)
+
+        # Compute SHA256 checksum
+        sha256 = compute_sha256(target_path)
+
+        # Open with rasterio and get bounds
+        with rasterio.open(str(target_path), "r") as src:
+            bounds = src.bounds
+            transformed_bounds = rasterio.warp.transform_bounds(
+                src.crs, "EPSG:4326", *bounds
+            )
+
+        # Create dataset entry
+        dataset = create_dataset_entry(
+            filename, target_path, sha256, transformed_bounds, user.id, copy_time
+        )
+
+        # Clean up temporary files
+        for chunk_file in temp_dir.glob("chunk_*"):
+            chunk_file.unlink()
+        temp_dir.rmdir()
```

Comment:
Can we add an `assert` / sanity check that temp_dir is not equal to other relevant dirs, that is cog, archive, thumbs?

Reply:
Hmm ok. rm is always dangerous:)